performance_curves/performance_curve_bundle.py

```diff
@@ -0,0 +1,54 @@
+from performance_curves.performance_curve import *
+
+
+class PerformanceCurveBundle:
+    def __init__(self,
+                 y_true: np.ndarray,
+                 y_scores: List[np.ndarray],
+                 metric: Metric,
+                 model_names: List[str],
+                 num_bins: Optional[int] = None,
+                 order_descending: bool = True,
+                 random_model: bool = True,
+                 random_seed: Optional[int] = None,
+                 num_trials: Optional[int] = 1,
+                 perfect_model: bool = True):
+        self.y_true = y_true
+        self.y_scores = y_scores
+        self.metric = metric
+        self.model_names = model_names
+        self.num_bins = num_bins
+
+        score_sizes = [len(y_score) for y_score in self.y_scores]
+        assert all(score_size == score_sizes[0] for score_size in score_sizes), \
+            "Number of predictions outputted by each model must be the same."
+
+        all_performance_values = []
+        all_case_counts = []
+        for i in range(len(self.y_scores)):
+            individual_curve = PerformanceCurve(self.y_true, y_scores[i], self.metric, self.num_bins,
+                                                order_descending)
+            all_performance_values.append(individual_curve.performance_values)
+            all_case_counts.append(individual_curve.case_counts)
+        self.bundle = dict(zip(self.model_names, zip(all_performance_values, all_case_counts)))
+
+        if random_model:
+            random_curve = RandomPerformanceCurve(self.y_true, self.metric, num_trials, self.num_bins, random_seed)
+            self.bundle["random_model"] = tuple(zip(random_curve.performance_values, random_curve.case_counts))
+        if perfect_model:
+            perfect_curve = PerfectPerformanceCurve(self.y_true, self.metric, self.num_bins)
+            self.bundle["perfect_model"] = tuple(zip(perfect_curve.performance_values, perfect_curve.case_counts))
```

Comment:
It seems likely that the random and perfect curves would always be handled as a special case. What do you think about putting them in their own attributes (not the `bundle`/`curves` dict) so that we don't need repeatedly check the key to make sure it isn't the random or perfect.

____

Separately, if we keep the dictionary then we should try to make keys we control ("random_model", "perfect_model") a bit more unlikely to conflict with the names the user passed in. I would add  one or two `_` to the beginning of the names we set.

Reply:
Making `random_curve` and `perfect_curve` as part of the `PerformanceCurveBundle` class, but not part of the `curves` dictionary (attribute of `PerformanceCurveBundle`) indeed makes the code much cleaner! I confused myself at first by naming the attribute `bundle`, but renaming it makes the concept clearer. 

Cause we now separate `random_curve` and `perfect_curve` from the `curves` attribute, we don't need to worry about potential name conflicts with what the user passes in anymore.